AK/Math.h

```diff
@@ -765,14 +766,21 @@ constexpr T log2(T x)
 
     // FIXME: Handle denormalized numbers separately
 
-    FloatExtractor<T> mantissa_ext {
-        .mantissa = ext.mantissa,
-        .exponent = FloatExtractor<T>::exponent_bias,
-        .sign = ext.sign
-    };
-
     // (1 <= mantissa < 2)
-    T m = mantissa_ext.d;
+    T m;
+    if constexpr (HostIsLittleEndian) {
+        m = ((FloatExtractor<T>){
+            .mantissa = ext.mantissa,
+            .exponent = FloatExtractor<T>::exponent_bias,
+            .sign = ext.sign
+        }).d;
+    } else {
+        m = ((FloatExtractor<T>){
+            .sign = ext.sign,
+            .exponent = FloatExtractor<T>::exponent_bias,
+            .mantissa = ext.mantissa
+        }).d;
+    }
```

Comment:
Seems fine to me too overall, but this thing makes me wonder if `FloatExtractor` should have been `struct` in the first place. Even before this PR, we had "theoretical" issues with `-m32 -mlong-double-128`, then there were problems with bitfields on `x86_64-pc-windows-msvc`. And now every single usage should be `if constexpr (HostIsLittleEndian)`

Reply:
I mean. If we add a constructor it won't need to be guarded right? Or is there some reason that it should remain an aggregate?

Reply:
I guess, ideally, it should be
```cpp
template<typename T>
struct FloatComponents;

template<>
struct FloatComponents<f32> {
    int mantissa;
    int exponent;
    bool sign;
};

...

template<FloatingPoint T>
FloatComponents<T> decompose_floating_point(T);

template<FloatingPoint T>
T assemble_floating_point(FloatComponents<T>);
```

This way we can also hide idiosyncrasy of Intel's f80 from the rest of the code.

Reply:
A constructor should definitely work to hide the if/else.

But would a struct? Can structs in C++ be initialized with random field order?
This is the only reason I added the if/else, because the compiler only allows specifying the fields in their definition order.

Reply:
Let's leave the best API bikeshedding for a follow-up :)